tests/test_events/handler/addTab.test.js

```diff
@@ -0,0 +1,138 @@
+    test("Should be skipped if repo does not exist",() => {
+        const handler = new tabEventHandler();
+        handler.handleTabChangeEvent();
+        // Verify no tab file should be generated
+        assertNoTabsRecorded(tabsRepo)   
+        });
+
+    //  Skips cases where repo exists, but is not connected
+    test("Should skip if repo exists, but is not connected", () => {
+        const handler = new tabEventHandler(repoPath);
+        handler.handleTabChangeEvent();
+        // Verify no tab file should be generated
+        assertNoTabsRecorded(tabsRepo)    
+    })
+
+    // Skip case where repo is connected, but user account is not valid
+    test("Should skip if repo is connected, but user account is not valid", () => {
+        configUtil.addRepo();
+        addUser(baseRepoPath, false);
+        const handler = new tabEventHandler(repoPath);
+        handler.handleTabChangeEvent(false);
+        // Verify no tab file should be generated
+        assertNoTabsRecorded(tabsRepo)       
+    })
+
+    // Skip case where user account is not valid, but repo is not connected 
+    test("Should skip if user account is valid, but repo is not connected", () => {
+        addUser(baseRepoPath, true);
+        const handler = new tabEventHandler(repoPath);
+        handler.handleTabChangeEvent();
+        // Verify no tab file should be generated
+        assertNoTabsRecorded(tabsRepo)      
+    })
+
+    // Should skip if not opened/closed event
+    test("Should skip if not opened/closed event", () => {
+        // Repo is connected & user account is also valid
+        configUtil.addRepo();
+        addUser(baseRepoPath, true);
+        const handler = new tabEventHandler(repoPath);
+        // Pass isTabEvent=false, which means changeEvent.changed.length > 0
+        handler.handleTabChangeEvent(false);
+        // Verify no tab file should be generated
+        assertNoTabsRecorded(tabsRepo)    
+    })
+
+    // Skip if invalid repo id
+    test("Should skip if repoId doesn't exist", () => {
+        const invalid_repo_path = (repoPath).slice(0,5);
+        const configUtil_2 = new Config(invalid_repo_path, configPath);
+        addUser(baseRepoPath, true);
+        const handler = new tabEventHandler(invalid_repo_path);
+        handler.handleTabChangeEvent(true);
+        // Verify no tab file should be generated
+        assertNoTabsRecorded(tabsRepo)    
+    })
+
+    test("Tabs data in positive case", async () => {
+        const mockTabGroups = [
+            { 
+                tabs: [
+                    { file_id: 1, path: `${newFilePath1}` },
+                    { file_id: 2, path: `${newFilePath2}` },
+                ],
+                activeTab: { id: 1 } 
+            }
+        ];
+        const spy = jest.spyOn(vscode.window.tabGroups, 'all').mockReturnValue(mockTabGroups);    
+        const result = vscode.window.tabGroups.all();
+        expect(result).toEqual(mockTabGroups);
```

Comment:
This seems we are asserting if `spyOn` is working correctly or not, we don't need this I think